### PR TITLE
Adds a new chemical, Saltwater. It induces puking when inside the body.

### PR DESCRIPTION
**Pull request by Irradiation**
Description: "It's water mixed with salt. It's probably not healthy to drink."
Recipe: 50 parts Water, 5 parts Sodium Chloride (Table Salt).
Result: 50 units.
Metabolism: 0.2

When inside your body, saltwater will have a 20% chance per tick to cause you to vomit and deal a minor amount of toxin damage.

Mainly used as a ghetto replacement for purging chems in the same vein as Activated Charcoal, except it's less effective and will mildly damage your body. Can also be used to challenge your fellow greyshirts to a puking contest!

EDIT 1: Changed the recipe (1u Water + 1u SodiumChloride into 50u Water + 5u Sodium Chloride), lowered the toxin damage (4 toxin down to 1 toxin (2 * REM)), changed the density and specheatcap.
EDIT 2: YOU'LL PUKE YOUR STEAK AND YOU WILL LIKE IT @Probe1 

:cl:
* rscadd: Added Saltwater. It induces puking when inside the body.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -6400,4 +6400,24 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
...
+	
+/datum/reagent/saltwater/on_mob_life(var/mob/living/M)
+
+	if(..())
+		return 1
+
+	if(ishuman(M) && prob(20))
+		var/mob/living/carbon/human/H = M
+		H.vomit()
+		M.adjustToxLoss(4)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Reagents.dm]: Have you tested these numbers?

This is equally as toxic as bleach fumes, and more toxic than plasma.

Times it by REM for consistency

**Comment by Shadowmech88** [comment]
Since it would be both feasible in-game and technically accurate, I think 1 or 2 parts salt to 10 parts water would be a better recipe.

**Comment by Irradiation** [comment]
Regarding the toxin damage, vomit heals 3 toxins when it happens. The idea of 4 toxins was to allow for 1 damage to flow through.

**Comment by Irradiation** [comment]
Regarding the toxin damage, vomit heals 3 toxins when it happens. The idea of 4 toxins was to allow for 1 damage to flow through.

**Comment by SarahJohnson489** [comment]
Praise saltwater meme cryo

**Comment by DrCelt** [comment]
image

**Comment by DrCelt** [comment]
how do you plead

**Comment by Kammerjunk** [comment]
But vomit uses trans_to() which doesn't dupe, does it?

**Comment by Kammerjunk** [comment]
Oh it does, but only if you override the multiplier, which vomit does not do

**Comment by Irradiation** [comment]
@DrCelt
I plead non-guilty to that allegation. The main idea behind it was indeed a ghetto chem purge, not another way to dupe chems.  Even if it's a possibility, I'm fairly certain that chem duping is already a rare occurrence currently.

**Comment by unid15** [comment]
Can the reaction occur inside a human body?

**Comment by Irradiation** [comment]
@unid15 The "reaction" should happen in humans, in exactly the same way as Tonio, Chifir, Charcoal and a handful of other vomit-inducing chemicals do.

**Comment by Probe1** [comment]
image

Eating two nice steaks with some water shouldn't create saltwater inside your body.  Fix that by making it not react inside mobs.

**Comment by Probe1** [comment]
FIX IT

**Comment by Irradiation** [comment]
@Probe1 IT'S A FEAAAAATUUUURE

**Comment by Probe1** [comment]
God bless.

**Comment by Kammerjunk** [comment]
You know, wouldn't required_container have worked for this

**Comment by Kammerjunk** [comment]
Nope, doesn't check for subtypes

**Comment by Probe1** [comment]
That was my thought but after seeing him struggle with it I decided the easy route would serve.

**Comment by MadmanMartian** [comment]
gotta struggle to get good at coding

**Pull request merged**